File: java/leetcode/src/main/java/com/leetcode/BST/Convert_Sorted_Array_to_Binary_Search_Tree_108.java
package com.leetcode.BST;

public class Convert_Sorted_Array_to_Binary_Search_Tree_108 {
    public TreeNode sortedArrayToBST(int[] nums) {
        TreeNode root = createBST(nums, 0, nums.length - 1);
        return root;
    }

    TreeNode createBST(int[] nums, int lo, int hi) {
        if (lo > hi) return null;
        int mid = (lo + hi) / 2;
        TreeNode node = new TreeNode(nums[mid]);
        node.left = createBST(nums, lo, mid - 1);
        node.right = createBST(nums, mid + 1, hi);
        return node;
    }
}

File: js/algrithm/BST/Convert_Sorted_Array_to_Binary_Search_Tree_108.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
var sortedArrayToBST = function(nums) {
    const createBST = function(nums, start, end) {
        if (start > end) return null
        let mid = Math.floor((start + end) / 2)
        let node = new TreeNode(nums[mid])
        node.left = createBST(nums, start, mid - 1)
        node.right = createBST(nums, mid + 1, end)
        return node
    }

    const root = createBST(nums, 0, nums.length - 1)
    return root
};